src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubSubscriberUtil.java

```diff
@@ -148,8 +148,8 @@ public static void createSubscription(BooleanSupplier preCheck, BackoffConfig ba
 
     while (preCheck.getAsBoolean() && attempts-- > 0) {
 
-      try (SubscriptionAdminClient subscriptionAdminClient = clientSupplier.get()) {
-
+      try {
+        SubscriptionAdminClient subscriptionAdminClient = clientSupplier.get();
```

Comment:
Does this client need to be manually closed after the request ends?

Reply:
Yes. The Receiver will close it immediately, and the PubSubDirectDStream will re-use it for other operations as well and close only in the stop() method.